transport/src/behaviour/pubsub.rs

```diff
@@ -115,16 +105,17 @@ impl PubsubBehaviour {
             Some(x) => x,
             None => return Err("message with unknown topic"),
         };
-        if let MsgOrdering::Ordered { sequence_numbers } = &mut topic_state.msg_ordering {
-            let last_seq_no = sequence_numbers.get(&peer_id).copied().unwrap_or_default();
-            match msg.sequence_number {
-                None => return Err("message without sequence number"),
-                // Sequence numbers should be timestamp-based, can't be from the future
-                Some(seq_no) if seq_no > timestamp_now() => return Err("invalid sequence number"),
-                Some(seq_no) if seq_no <= last_seq_no => return Err("old message"),
-                Some(seq_no) => sequence_numbers.insert(peer_id, seq_no),
-            };
-        }
+        let last_seq_no = topic_state.sequence_numbers.entry(peer_id).or_default();
+        match msg.sequence_number {
+            None => return Err("message without sequence number"),
+            // Sequence numbers should be timestamp-based, can't be from the future
+            Some(seq_no) if seq_no > timestamp_now() => return Err("invalid sequence number"),
```

Comment:
Does `timestamp_now` recalc current time on each iteration? We should calculate it once because it's a syscall and gives different results